CRM/Event/Form/Registration.php

```diff
@@ -196,7 +196,7 @@ class CRM_Event_Form_Registration extends CRM_Core_Form {
    */
   public function preProcess() {
     $this->_eventId = CRM_Utils_Request::retrieve('id', 'Positive', $this, TRUE);
-    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE);
+    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'add');
```

Comment:
i feel like this should probably be changed to alphanumeric.

Reply:
You mean:
  ```$this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, CRM_Core_Action::ADD)``` ?

I did a grep on CRM codebase and all such occurrences I passing string value as default. Do a ```
$ grep -irn --color "CRM_Utils_Request::retrieve('action'" CRM
```

Reply:
Oh ok. I think it would be best to have:
$this->_action = CRM_Utils_Request::retrieve('action', 'Alphanumeric', $this, FALSE, CRM_Core_Action::ADD);

Reply:
Yep happy with that :+1: just figure Alphanumeric is bit more strict than string

Reply:
@seamuslee001 nope changing that to alphanumeric won't solve the issue as I tested on my local. When I hit ```www.example.com/civicrm/event/register?id=3&pcpId=2&reset=1&action=1``` it considers action value as string not integer and eventually it return ```0 = CRM_Core_Action::resolve($value)``` at CRM/Utils/Request.php at line 130. 

So I would say declaring Alhanumeric doesn't mean it will allow either string or integer value but it consider the datatype as string. As the name suggest alphanumeric means the datatype is String and it will allow string w/o integer value. 

So I think current format is ok or atleast I can change it to:
```
$this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, CRM_Core_Action::ADD);
```